### PR TITLE
Manual cherrypick of #15168

### PR DESCRIPTION
**Pull request by craigbox**
Takes the changes to the Getting Started guide, as well as some follow-up PRs and related issues, to `release-1.22`.

Hopefully.

### CODE REVIEW

File: content/en/docs/ambient/getting-started/test.sh [hunk 1]
```diff
@@ -15,67 +15,62 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# @setup profile=none
+# @setup profile=ambient
```
Comment (kfaseela) [inline content/en/docs/ambient/getting-started/test.sh:18]:
```suggestion
# @setup profile=none
```

File: content/en/docs/ambient/getting-started/test.sh [hunk 2]
```diff
@@ -15,67 +15,62 @@
...
 
-# Kubernetes Gateway API CRDs are required by waypoint proxy.
-snip_download_and_install_2
+source "content/en/docs/ambient/getting-started/deploy-sample-app/snips.sh"
+source "content/en/docs/ambient/getting-started/secure-and-visualize/snips.sh"
+source "content/en/docs/ambient/getting-started/enforce-auth-policies/snips.sh"
+source "content/en/docs/ambient/getting-started/manage-traffic/snips.sh"
+source "content/en/docs/ambient/getting-started/cleanup/snips.sh"
 
-# install istio with ambient profile
```
Comment (kfaseela) [inline content/en/docs/ambient/getting-started/test.sh:27]: we will have to retain the installation if we go with none profile.. else let me know we can even add an ambient profile CI job for 1.22 if that is what is needed

Comment (craigbox) [reply content/en/docs/ambient/getting-started/test.sh:27]: I don't know if we want the two branches to diverge to the point where they have different tests.
If it's possible, I think having the ambient profile CI in 1.22 is the easiest way out

**Comment by kfaseela** [comment]
> @craigbox: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> doc.test.profile-ambient_istio.io_release-1.22	[b79c016](https://github.com/istio/istio.io/commit/b79c016eb7f7a393082b5c9a9cd057b204e71ed1)	[link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15346/doc.test.profile-ambient_istio.io_release-1.22/1806366252256464896)	false	`/test doc.test.profile-ambient`

pattern /home/prow/go/src/istio.io/istio.io/tests/setup/profile-ambient/...: lstat /home/prow/go/src/istio.io/istio.io/tests/setup/profile-ambient/: no such file or directory

you need to cherrypick https://github.com/istio/istio.io/pull/15210 too

**Comment by kfaseela** [comment]
> > @craigbox: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> > Test name	Commit	Details	Required	Rerun command
> > doc.test.profile-ambient_istio.io_release-1.22	[b79c016](https://github.com/istio/istio.io/commit/b79c016eb7f7a393082b5c9a9cd057b204e71ed1)	[link](https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/15346/doc.test.profile-ambient_istio.io_release-1.22/1806366252256464896)	false	`/test doc.test.profile-ambient`
> 
> pattern /home/prow/go/src/istio.io/istio.io/tests/setup/profile-ambient/...: lstat /home/prow/go/src/istio.io/istio.io/tests/setup/profile-ambient/: no such file or directory
> 
> you need to cherrypick #15210 too

I just did it https://github.com/istio/istio.io/pull/15359

**Comment by kfaseela** [comment]
it would be better to wait for the wasm pr to be merged, and rebase this PR on top of that

**Comment by craigbox** [comment]
The ambient profile was added in https://github.com/istio/test-infra/pull/5410.
We are now failing on needing a script that was fixed on `master` in https://github.com/istio/istio/pull/51329.

Is there any way to resolve the test version to a commit on `release-1.22` or do we have to wait for 1.22.3 for this to potentially pass?

@dhawton @kfaseela

**Comment by kfaseela** [comment]
> The ambient profile was added in [istio/test-infra#5410](https://github.com/istio/test-infra/pull/5410). We are now failing on needing a script that was fixed on `master` in [istio/istio#51329](https://github.com/istio/istio/pull/51329).
> 
> Is there any way to resolve the test version to a commit on `release-1.22` or do we have to wait for 1.22.3 for this to potentially pass?
> 
> @dhawton @kfaseela

The script you have mentioned is fixing only a cleanup for bookinfo.. the test is failing to deploy bookinfo...

```
gateway.gateway.networking.k8s.io/bookinfo-gateway created
httproute.gateway.networking.k8s.io/bookinfo created
Error from server (NotFound): deployments.apps "bookinfo-gateway-istio" not found
Failed rollout of deployment bookinfo-gateway-istio in namespace default
```

File: content/en/docs/ambient/getting-started/test.sh [hunk 3]
```diff
@@ -15,67 +15,62 @@
...
 
 # @cleanup
-snip_uninstall_1
-snip_uninstall_2
-snip_uninstall_3
+snip_remove_the_ambient_and_waypoint_labels_1
+snip_remove_waypoint_proxies_1
+snip_remove_the_sample_application_1
 samples/bookinfo/platform/kube/cleanup.sh
-snip_uninstall_4
```
Comment (kfaseela) [inline content/en/docs/ambient/getting-started/test.sh:81]: you can add an explicit `kubectl delete authorizationpolicies --all` here as a workaround

File: content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh [hunk 4]
```diff
@@ -91,8 +87,8 @@ _verify_same snip_verify_the_traffic_targeting_the_service_3 "$snip_verify_the_t
 
 # @cleanup
 snip_cleanup_1
-snip_uninstall_1
-snip_uninstall_2
-snip_uninstall_3
+snip_remove_the_ambient_and_waypoint_labels_1
+snip_remove_waypoint_proxies_1
+snip_remove_the_sample_application_1
 samples/bookinfo/platform/kube/cleanup.sh
```
Comment (kfaseela) [inline content/en/docs/ambient/usage/extend-waypoint-wasm/test.sh:93]:
```suggestion
samples/bookinfo/platform/kube/cleanup.sh
kubectl delete authorizationpolicies --all
```

File: content/en/docs/ambient/getting-started/test.sh [hunk 5]
```diff
@@ -15,67 +15,62 @@
...
 "
 
 # @cleanup
-snip_uninstall_1
-snip_uninstall_2
-snip_uninstall_3
+snip_remove_the_ambient_and_waypoint_labels_1
+snip_remove_waypoint_proxies_1
+snip_remove_the_sample_application_1
 samples/bookinfo/platform/kube/cleanup.sh
```
Comment (kfaseela) [inline content/en/docs/ambient/getting-started/test.sh:75]:
```suggestion
samples/bookinfo/platform/kube/cleanup.sh
kubectl delete authorizationpolicy productpage-viewer
```

**Comment by kfaseela** [comment]
@craigbox all good now :)

**Review by craigbox — COMMENTED** [review]
LGTM, thanks @kfaseela !
Feel free to merge when you're ready

**Review by kfaseela — APPROVED** [review]

**Pull request merged**